File: src/agent_showdown/interfaces/game/obstacle.py
from pydantic import BaseModel, ConfigDict

from agent_showdown.interfaces.game.position import Position
from agent_showdown.interfaces.game.terrain_kind import TerrainKind


class Obstacle(BaseModel):
    """One blocked square. Nothing walks onto it and no bolt flies through it."""

    model_config = ConfigDict(frozen=True)

    position: Position
    kind: TerrainKind

File: web_client/src/interfaces/game/obstacle.ts
import type { Position } from "./position.js";

/** What stands on an obstructed square. Mirrors `TerrainKind` on the Python side. */
export type TerrainKind = "tree" | "stone_wall" | "boulder" | "stone_well";

/** One blocked square: nothing walks onto it and no bolt flies through it. */
export interface Obstacle {
  readonly position: Position;
  readonly kind: TerrainKind;
}
